src/main/java/org/gridsuite/network/map/dto/utils/ElementUtils.java

```diff
@@ -0,0 +1,229 @@
+        return Double.isNaN(d) ? null : d;
+    }
+
+    public static CurrentLimitsData toMapDataCurrentLimits(CurrentLimits limits) {
+        CurrentLimitsData.CurrentLimitsDataBuilder builder = CurrentLimitsData.builder();
+        boolean empty = true;
+        if (!Double.isNaN(limits.getPermanentLimit())) {
+            builder.permanentLimit(limits.getPermanentLimit());
+            empty = false;
+        }
+        if (!CollectionUtils.isEmpty(limits.getTemporaryLimits())) {
+            builder.temporaryLimits(toMapDataTemporaryLimit(limits.getTemporaryLimits()));
+            empty = false;
+        }
+        return empty ? null : builder.build();
+    }
+
+    private static List<TemporaryLimitData> toMapDataTemporaryLimit(Collection<LoadingLimits.TemporaryLimit> limits) {
+        return limits.stream()
+                .map(l -> TemporaryLimitData.builder()
+                        .name(l.getName())
+                        .acceptableDuration(l.getAcceptableDuration() == Integer.MAX_VALUE ? null : l.getAcceptableDuration())
+                        .value(l.getValue() == Double.MAX_VALUE ? null : l.getValue())
+                        .build())
+                .collect(Collectors.toList());
+    }
+
+    public static String getBusOrBusbarSection(Terminal terminal) {
+        String busOrBusbarSectionId;
+        if (terminal.getVoltageLevel().getTopologyKind().equals(TopologyKind.BUS_BREAKER)) {
+            if (terminal.isConnected()) {
+                busOrBusbarSectionId = terminal.getBusBreakerView().getBus().getId();
+            } else {
+                busOrBusbarSectionId = terminal.getBusBreakerView().getConnectableBus().getId();
+            }
+        } else {
+            busOrBusbarSectionId = getBusbarSectionId(terminal);
+        }
+        return busOrBusbarSectionId;
+    }
+
+    public static String getBusbarSectionId(Terminal terminal) {
+        BusbarSectionFinderTraverser connectedBusbarSectionFinder = new BusbarSectionFinderTraverser(terminal.isConnected());
+        terminal.traverse(connectedBusbarSectionFinder);
+        return connectedBusbarSectionFinder.getFirstTraversedBbsId();
+    }
+
+    public static List<TapChangerStepData> toMapDataPhaseStep(Map<Integer, PhaseTapChangerStep> tapChangerStep) {
+        if (tapChangerStep == null) {
+            return List.of();
+        }
+        return tapChangerStep.entrySet().stream().map(p -> {
+            Integer index = p.getKey();
+            PhaseTapChangerStep v = p.getValue();
+
+            return TapChangerStepData.builder().index(index)
+                    .g(v.getG())
+                    .b(v.getB())
+                    .r(v.getR())
+                    .x(v.getX())
+                    .rho(v.getRho())
+                    .alpha(v.getAlpha())
+                    .build();
+        }).collect(Collectors.toList());
+    }
+
+    public static TapChangerData toMapData(PhaseTapChanger tapChanger) {
```

Comment:
```suggestion
    public static TapChangerData toMapDataTapChanger(PhaseTapChanger tapChanger) {
```
To be more coherent with the rest of the class